cpp/harvest/core.cc

```diff
@@ -553,6 +556,95 @@ void DataProviders(int argc, char **argv) {
 }
 
 
+void SplitDataProviderId(int argc, char **argv) {
+    if (argc != 4)
+        Usage();
+
+    std::cout << "Preparing data..." << std::endl;
+    const auto works(CORE::GetWorksFromFile(argv[2]));
+
+    std::vector<unsigned> list_of_data_provider_id;
+    std::map<std::string, unsigned> data_provider_id_counter;
+    unsigned total_record_counter(0), progress(0), step(0), display_next(0), closed_counter(0);
+    std::string item, outputFile, outputDir(argv[3]);
+
+
+    std::cout << "Processing " << works.size() << " records ..." << std::endl;
+    for (auto work : works) {
+        if (!work.getDataProviderIds().empty()) {
+            for (auto data_provider_id : work.getDataProviderIds()) {
+                item = std::to_string(data_provider_id);
+                outputFile = outputDir + item + ".json";
+                if (std::find(list_of_data_provider_id.begin(), list_of_data_provider_id.end(), data_provider_id)
+                    != list_of_data_provider_id.end()) {
+                    // id is exist in the list
+                    CORE::OutputFileAppend(outputFile, work, 0);
+                    ++data_provider_id_counter[item];
+                } else {
+                    // a new unique id
+                    list_of_data_provider_id.emplace_back(data_provider_id);
+                    data_provider_id_counter[item] = 1;
+                    CORE::OutputFileStart(outputFile);
+                    CORE::OutputFileAppend(outputFile, work, 1);
+                }
+            }
+        }
+
+        ++total_record_counter;
+
+        // displaying progress
+        progress = (100 * (total_record_counter)) / works.size();
+        if (progress >= display_next) {
+            std::cout << "\r"
+                      << "[" << std::string(progress / 5, (char)35u) << std::string(100 / 5 - progress / 5, ' ') << "]";
+            std::cout << progress << "%"
+                      << " [record " << total_record_counter << " of " << works.size() << "]";
+            std::cout.flush();
+            display_next += step;
```

Comment:
Does `step` ever change? It always seems to be 0.

Reply:
it is just a constant. 0 means will display the progress percentage every time it changes. Ex. the step is five means it will show the progress percentage every 5% of progress. 
Maybe I will change its type to const unsigned.